project/threads/views.py

```diff
@@ -25,6 +24,21 @@
 )
 
 
+class ThreadDetailView(LoginRequiredMixin, DetailView):
+    model = Thread
+    context_object_name = "thread"
+    template_name = "thread.html"
+    login_url = "accounts_login"
+
+    def get_context_data(self, **kwargs):
+        context = super(ThreadDetailView, self).get_context_data(**kwargs)
+        context["categories"] = Category.objects.all()
+        context["problems"] = Civi.objects.problems(thread_id=context["thread"].id)
+        context["causes"] = Civi.objects.causes(thread_id=context["thread"].id)
+        context["solutions"] = Civi.objects.solutions(thread_id=context["thread"].id)
```

Comment:
Rather than explicitly attach these to context, let's add them as properties directly on the `Thread` model. Then, the template or other code could read as follows.

- `thread.problems`
- `thread.causes`
- `thread.solutions`

Reply:
Good point 👍 I will fix them.